### PR TITLE
feat: nft gallery UI

### LINKED ISSUE #1380: NFT gallery

### PR DESCRIPTION
**Pull request by sasicodes**
## What does this PR do?

Closes #1380 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't read the [contributing guide](https://github.com/lensterxyz/lenster/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/src/components/NFT/SingleNFT.tsx [hunk 1]
```diff
@@ -1,70 +1,27 @@
...
-      </div>
-    </Card>
+    <div className="p-2">
+      <img
+        className="rounded-xl h-[260px] object-cover"
+        src={`${
+          nft.originalContent.uri
+            ? getIPFSLink(nft.originalContent.uri)
+            : `${STATIC_IMAGES_URL}/placeholder.webp`
+        }`}
```
Comment (harish-sethuraman) [inline apps/web/src/components/NFT/SingleNFT.tsx:19]:
```suggestion
        src={
          nft.originalContent.uri
            ? getIPFSLink(nft.originalContent.uri)
            : `${STATIC_IMAGES_URL}/placeholder.webp`
        }
```

File: apps/web/src/components/Profile/NftGallery/Gallery.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  nfts: Nft[];
+};
+
+const Gallery: FC<Props> = ({ nfts }) => {
+  const [items, setItems] = useState(nfts.filter((nft) => !nft?.originalContent?.animatedUrl));
+
+  function onChange(sourceId: any, sourceIndex: any, targetIndex: any) {
+    const result = swap(items, sourceIndex, targetIndex);
+    console.log('🚀 ~ file: Gallery.tsx:20 ~ onChange ~ result', result);
+    return setItems([...result]);
```
Comment (harish-sethuraman) [inline apps/web/src/components/Profile/NftGallery/Gallery.tsx:16]:
```suggestion
    return setItems(result);
```
aren't result array already?

**Comment by bigint** [comment]
T-123

**Comment by bigint** [comment]
Link T-123

**Comment by bigint** [comment]
Link T-2572

**Comment by bigint** [comment]
@sasicodes can you fix conflicts 🙇🏼

File: apps/web/package.json [hunk 3]
```diff
@@ -43,6 +44,8 @@
     "plyr-react": "^5.1.0",
     "react": "^18.2.0",
     "react-copy-to-clipboard": "^5.1.0",
+    "react-dnd": "^16.0.1",
+    "react-dnd-html5-backend": "^16.0.1",
```
Comment (bigint) [inline apps/web/package.json:48]: @harish-sethuraman are we ok with this lib? tagging you cuz u are expert in this 😉

Comment (harish-sethuraman) [reply apps/web/package.json:48]: If we have vertical collection, sorting and searching then we can go with `react-beautiful-dnd` if that cannot handle what you want then react dnd is fine as well.

**Comment by bigint** [comment]
Link T-2572

File: apps/web/src/components/Profile/NftGallery/NoGallery.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  return (
+    <div className="py-10 flex space-y-4 flex-col items-center justify-center">
+      <Create showModal={showCreateModal} setShowModal={setShowCreateModal} />
+      <div className="grid grid-cols-3 w-full max-w-sm gap-4">
+        <div className="col-span-2 flex h-[250px] justify-center items-center bg-brand-100 rounded-xl">
+          <img
+            width={100}
+            height={100}
+            className="w-20 h-20"
+            src="https://i.imgur.com/OAauOym.png"
```
Comment (sasicodes) [inline apps/web/src/components/Profile/NftGallery/NoGallery.tsx:29]: @bigint we need to upload these images to assets 🙇🏻

Comment (bigint) [reply apps/web/src/components/Profile/NftGallery/NoGallery.tsx:29]: sure will upload this and share link here

File: packages/data/feature-flags.ts [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import { mainnetStaffs } from 'staffs';
```
Comment (bigint) [inline packages/data/feature-flags.ts]: It should be `./staffs` i guess @sasicodes

Comment (sasicodes) [reply packages/data/feature-flags.ts]: check the latest commit 🙇🏻

File: apps/web/src/components/Profile/NftGallery/NoGallery.tsx (hunk 4 above)
Comment (sasicodes) [reply apps/web/src/components/Profile/NftGallery/NoGallery.tsx:29]: Reminder

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import { Modal } from '@components/UI/Modal';
+import { ChevronLeftIcon } from '@heroicons/react/outline';
+import { Trans } from '@lingui/macro';
+import clsx from 'clsx';
+import type { Dispatch, FC } from 'react';
+import React, { useState } from 'react';
+
+import Picker from './Picker';
+
+type Props = {
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:12]: interface bls, cuz we are using that everywhere

File: apps/web/src/components/Profile/NftGallery/Create.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  showModal: boolean;
+  setShowModal: Dispatch<boolean>;
+};
+
+const Create: FC<Props> = ({ showModal, setShowModal }) => {
+  const [emoji, setEmoji] = useState('');
+  const [pickNfts, setPickNfts] = useState(false);
+
+  return (
+    <div>
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Create.tsx:22]: do we need div here? or just fragment `<>`?

File: apps/web/src/components/Profile/NftGallery/Detail/index.tsx [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+import formatHandle from '@lib/formatHandle';
+import getAvatar from '@lib/getAvatar';
+import isFeatureEnabled from '@lib/isFeatureEnabled';
+import type { Profile } from 'lens';
+import Link from 'next/link';
+import React from 'react';
+import Custom404 from 'src/pages/404';
+import { useAppStore } from 'src/store/app';
+
+const NFTDetail = () => {
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Detail/index.tsx:15]: FC

**Review by bigint — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Profile/NftGallery/Create.tsx (hunk 7 above)
Comment (sasicodes) [reply apps/web/src/components/Profile/NftGallery/Create.tsx:22]: not requried, removing

File: apps/web/src/components/Profile/NftGallery/Detail/index.tsx (hunk 8 above)
Comment (sasicodes) [reply apps/web/src/components/Profile/NftGallery/Detail/index.tsx:15]: FC psycho

File: apps/web/src/components/Profile/NftGallery/Gallery.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import type { Nft } from 'lens';
+import type { FC } from 'react';
+import React from 'react';
+
+import NftCard from './NftCard';
+
+type Props = {
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/Gallery.tsx:7]: interface

File: apps/web/src/components/Profile/NftGallery/NoGallery.tsx [hunk 10]
```diff
@@ -0,0 +1,83 @@
+import { Button } from '@components/UI/Button';
+import { Trans } from '@lingui/macro';
+import type { Profile } from 'lens';
+import type { FC } from 'react';
+import React, { useState } from 'react';
+import { useAppStore } from 'src/store/app';
+
+import Create from './Create';
+
+type Props = {
```
Comment (bigint) [inline apps/web/src/components/Profile/NftGallery/NoGallery.tsx:10]: interface

File: apps/web/src/components/Profile/NftGallery/NoGallery.tsx (hunk 4 above)
Comment (bigint) [reply apps/web/src/components/Profile/NftGallery/NoGallery.tsx:29]: I'll take care of it 🙇🏼

**Review by bigint — CHANGES_REQUESTED** [review]

**Comment by bigint** [comment]
**LGTM** :shipit: 

Rad LGTM

**Comment by sasicodes** [comment]
Closes T-123

**Pull request merged**